python-avd/pyavd/_eos_designs/structured_config/network_services/router_bgp.py

```diff
@@ -303,8 +303,9 @@ def _update_router_bgp_vrf_mlag_neighbor_cfg(self: AvdStructuredConfigNetworkSer
         """
         In-place update MLAG neighbor part of structured config for *one* VRF under router_bgp.vrfs
         """
-        if not self._mlag_ibgp_peering_redistribute(vrf, tenant):
-            bgp_vrf["redistribute_routes"][0]["route_map"] = "RM-CONN-2-BGP-VRFS"
+        if get(vrf, "redistribute_connected", default=True):
+            if not self._mlag_ibgp_peering_redistribute(vrf, tenant):
+                bgp_vrf["redistribute_routes"][0]["route_map"] = "RM-CONN-2-BGP-VRFS"
```

Comment:
Guillaume meant to move the check for `get(vrf, "redistribute_connected", default=True)` into the method `def _mlag_ibgp_peering_redistribute(...)` So anything else using this method will honor the same.

Reply:
Note that you must return True from that method in the case redistribute-connected is set to false since the check here has the NOT. Maybe we should reverse the logic and rename the method to _exclude_mlag_ibgp_peering_from_redistribute. Please analyze the logic and search for other uses of the method to see that you retain the current behavior.

Reply:
Fixed.

Reply:
There is a bug in the new logic. If you try to set the new value to false on the VRF Tenant_A_WEB_ZONE you should see an error around the line mentioned above.

Reply:
_exclude_mlag_ibgp_peering_redistribute was returning True when `redistribute_connected` was false. Now changed it to return false. This is not giving the error for Tenant_A_WEB_ZONE when redistribute_connected is set to false.

Reply:
Thanks.

Reply:
Added a new vrf in network services to test this condition.